backend/controller/vm.js

```diff
@@ -0,0 +1,130 @@
+const sshKeygen = require("ssh-keygen-lite");
+const sshpk = require("sshpk");
+const path = require("path");
+const { createAzureVm } = require("../utils/createVm");
+const { BadRequestError } = require("../errors");
+const {
+    ClientSecretCredential,
+    DefaultAzureCredential,
+} = require("@azure/identity");
+const { ComputeManagementClient } = require("@azure/arm-compute");
+const { ResourceManagementClient } = require("@azure/arm-resources");
+const { NetworkManagementClient } = require("@azure/arm-network");
+require("dotenv").config();
+const virtualMachineServiceId = "vm";
+
+const createVm = async (req, res) => {
+    const { serviceId, userAddr, duration } = req.body;
```

Comment:
is there a provision for the `signature` meant for authentication?

Reply:
yh